ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -171,6 +172,10 @@ class ( Show (ChainState    p)
   -- | We require that protocols support a @k@ security parameter
   protocolSecurityParam :: NodeConfig p -> SecurityParam
 
+  -- | We require that protocols support a magic number used during version
+  -- negotiation.
+  protocolNetworkMagic  :: NodeConfig p -> NetworkMagic
```

Comment:
I don't see where this is actually used.

Reply:
Discussed on slack. But `protocolNetworkMagic` is used by users of ouroboros-network such as cardano-node and cardano-explorer.